chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -407,84 +432,102 @@ func (b *BitcoindNotifier) handleRelevantTx(tx chain.RelevantTx, bestHeight int3
 // historicalConfDetails looks up whether a transaction is already included in a
 // block in the active chain and, if so, returns details about the confirmation.
 func (b *BitcoindNotifier) historicalConfDetails(txid *chainhash.Hash,
-	heightHint, currentHeight uint32) (*chainntnfs.TxConfirmation, error) {
+	heightHint, currentHeight uint32) (*chainntnfs.TxConfirmation, txConfStatus, error) {
 
-	// First, we'll attempt to retrieve the transaction details using the
-	// backend node's transaction index.
-	txConf, err := b.confDetailsFromTxIndex(txid)
-	if err != nil {
-		return nil, err
-	}
+	// We'll first attempt to retrieve the transaction using the node's
+	// txindex.
+	txConf, txStatus, err := b.confDetailsFromTxIndex(txid)
+
+	// We'll then check the status of the transaction lookup returned to
+	// determine whether we should proceed with any fallback methods.
+	switch {
+	// The transaction was found within the node's txindex.
+	case txStatus == txFoundMempool:
+
+	// The transaction was found within the node's txindex.
+	case txStatus == txFoundIndex:
 
-	if txConf != nil {
-		return txConf, nil
+	// The transaction was not found within the node's mempool or txindex.
+	case txStatus == txNotFoundIndex && err == nil:
```

Comment:
Should we return in the case of `err != nil`?

Reply:
This indicates that the transaction simply doesn't exist, as it was not found within the txindex or mempool. The `err != nil` case is handled in the `default` case, as that means something went wrong when looking at the index (either because it's disabled or something else went wrong), so we should fall back to scanning manually.